### PR TITLE
Adding V3 compatible windows integration

### PR DESCRIPTION
**Pull request by RagalahariP**
- Adding windows-daemonset.yaml file similar to V2 and supporting changes in templates.

Testing : Tested in local EKS cluster with the following node config

```
kubectl get nodes -o wide
NAME                          STATUS   ROLES    AGE     VERSION                INTERNAL-IP   EXTERNAL-IP      OS-IMAGE                         KERNEL-VERSION                 CONTAINER-RUNTIME
ip-10-0-11-231.ec2.internal   Ready    <none>   3d21h   v1.23.17-eks-a59e1f0   10.0.11.231   18.208.247.135   Windows Server 2022 Datacenter   10.0.20348.1668                docker://20.10.21
ip-10-0-110-39.ec2.internal   Ready    <none>   22d     v1.23.17-eks-a59e1f0   10.0.110.39   <none>           Windows Server 2019 Datacenter   10.0.17763.4252                docker://20.10.21
ip-10-0-27-190.ec2.internal   Ready    <none>   22d     v1.23.17-eks-a59e1f0   10.0.27.190   3.95.250.93      Amazon Linux 2                   5.4.238-148.347.amzn2.x86_64   docker://20.10.17

```

Dashboard:  https://onenr.io/0BQ18La75jx

### CODE REVIEW

File: charts/newrelic-infrastructure/templates/clusterrole.yaml
Comment (nrepai) [inline charts/newrelic-infrastructure/templates/clusterrole.yaml:21]: is this in place to fix existing issues with rbac? if so, this should just be a separate PR that we can merge without the fix being blocked by the windows work

File: charts/newrelic-infrastructure/templates/kubelet/daemonset.yaml
Comment (nrepai) [inline charts/newrelic-infrastructure/templates/kubelet/daemonset.yaml:255]: do we know how recent was this label introduced to k8s? would using it affect clusters using older versions?

File: charts/newrelic-infrastructure/values-windows-linux.yaml
Comment (nrepai) [inline charts/newrelic-infrastructure/values-windows-linux.yaml]: i couldn't find anywhere that uses this value in this PR. is this needed?

File: charts/newrelic-infrastructure/values-windows-linux.yaml
Comment (nrepai) [inline charts/newrelic-infrastructure/values-windows-linux.yaml]: i couldn't find anywhere that uses this value in this PR. is this needed?

File: charts/newrelic-infrastructure/templates/kubelet/daemonset-windows.yaml
Comment (nrepai) [inline charts/newrelic-infrastructure/templates/kubelet/daemonset-windows.yaml:3]: this is really good. i'm slightly concerned about how there are two daemnoset templates that might have diverging settings beyond just the os-specific configurations. for example, the windows daemonset is missing `updateStrategy` that exist in the linux daemonset.

it might be worth having the two share a common daemonset template to minimize this type of divergence. if you agree with this approach, i'd recommend:

1. PR first to extract `charts/newrelic-infrastructure/templates/kubelet/daemonset.yaml` into a template like `charts/newrelic-infrastructure/templates/kubelet/_daemonset.tpl` and pass in values to recreate the existing linux daemonset. the point of this PR is to ensure the existing linux daemonset has the same output as before
2. add the windows daemonset passing different values to `_daemonset.tpl`

so the two daemonsets can share some boilerplate

File: charts/newrelic-infrastructure/values-windows-linux.yaml
Comment (nrepai) [inline charts/newrelic-infrastructure/values-windows-linux.yaml]: should this be in `values.yaml`? i don't see it being imported anywhere

File: charts/newrelic-infrastructure/templates/kubelet/daemonset-windows.yaml
Comment (nrepai) [inline charts/newrelic-infrastructure/templates/kubelet/daemonset-windows.yaml]: it seems like we can just range over the values {{- range .Values.kubelet.windowsOsList }}

File: charts/newrelic-infrastructure/templates/clusterrole.yaml
Comment (nrepai) [reply charts/newrelic-infrastructure/templates/clusterrole.yaml:21]: from discussion: this seems to be specific to windows since it's still using infra agent v2. so let's add a conditional or at least code comment

File: charts/newrelic-infrastructure/templates/kubelet/daemonset-windows.yaml
Comment (nrepai) [reply charts/newrelic-infrastructure/templates/kubelet/daemonset-windows.yaml:4]: from discussion: there's a lot of dependencies between windows and linux daemonsets. so we will leave this cleanup work as follow up

File: charts/newrelic-infrastructure/values-windows-linux.yaml
Comment (RagalahariP) [reply charts/newrelic-infrastructure/values-windows-linux.yaml]: Removed this label.

File: charts/newrelic-infrastructure/values-windows-linux.yaml
Comment (RagalahariP) [reply charts/newrelic-infrastructure/values-windows-linux.yaml]: Yes, global values are required. The New Relic common library uses these values and set them globally (like LicenseKey)[Reference]( https://github.com/newrelic/nri-kubernetes/blob/main/charts/newrelic-infrastructure/README.md#affinities-and-tolerations )
This is an example yaml file added for windows integration.

File: charts/newrelic-infrastructure/templates/kubelet/daemonset-windows.yaml
Comment (RagalahariP) [reply charts/newrelic-infrastructure/templates/kubelet/daemonset-windows.yaml]: Done

File: charts/newrelic-infrastructure/templates/clusterrole.yaml
Comment (RagalahariP) [reply charts/newrelic-infrastructure/templates/clusterrole.yaml:21]: Adding more details here, 
V2 integration has “pod” in rbac resources. But its removed in V3. Since we are using windows integration same as in V2,  I have added “pod” in rbac.  With out this change, we get follwoing error from Windows agent
time="2023-05-12T10:30:40Z" level=error msg="Integration command failed" error="exit status 1" instance=nri-kubernetes integration=com.newrelic.kubernetes prefix=integration/com.newrelic.kubernetes stderr="time=\"2023-05-12T10:30:40Z\" level=panic msg=\"failed to discover nodeIP with kube-state-metrics, got error: pods is forbidden: User \\\"system:serviceaccount:newrelic:newrelic-newrelic-infrastructure\\\" cannot list resource \\\"pods\\\" in API group \\\"\\\" at the cluster scope\"\ntime=\"2023-05-12T10:30:40Z\" level=fatal msg=\"failed to discover nodeIP with kube-state-metrics, got error: pods is forbidden: User \\\"system:serviceaccount:newrelic:newrelic-newrelic-infrastructure\\\" cannot list resource \\\"pods\\\" in API group \\\"\\\" at the cluster scope\"\n" working-dir="C:\\Program Files\\New Relic\\newrelic-infra\\newrelic-integrations”

**Review by nrepai — APPROVED** [review]
lgtm. i dont have merge permission for this repo but also we should get a second pair of eyes on this. ill ping the team

**Comment by nrepai** [comment]
actually i just noticed the tests are failing. not sure if it's specific to this branch since I see other PRs failing

File: charts/newrelic-infrastructure/values-windows.yaml
Comment (htroisi) [inline charts/newrelic-infrastructure/values-windows.yaml:8]: Should we upstream `enableWindows` and `windowsOsList` in the chart's values.yaml file?

Comment (RagalahariP) [reply charts/newrelic-infrastructure/values-windows.yaml:8]: Updated

**Comment by htroisi** [comment]
@RagalahariP I DM'd you the instructions to get the E2E tests to pass.

File: charts/newrelic-infrastructure/values.yaml
Comment (htroisi) [inline charts/newrelic-infrastructure/values.yaml:133]:
```suggestion
# -- Enable monitoring on Windows nodes.
```

Comment (htroisi) [inline charts/newrelic-infrastructure/values.yaml:133]: Could you modify the comments for `enableWindows` and `enableWindows` to use the same formatting as the rest of the file? 
Once you've modified the formatting can you run `helm-docs` tool to automatically update the README.md file?

File: charts/newrelic-infrastructure/values-windows.yaml
Comment (htroisi) [reply charts/newrelic-infrastructure/values-windows.yaml:4]: @RagalahariP can you address this comment? I'm not sure this is needed.

Comment (RagalahariP) [reply charts/newrelic-infrastructure/values-windows.yaml]: @htroisi  Yes, I have deleted redundant values-windows.yaml . Now values.yaml has windows config option.

**Comment by htroisi** [comment]
Closing this PR is favor of #754 since E2E tests don't work from forks.

**Pull request closed without merging**
